src/LinuxTracing/UprobesUnwindingVisitorSampleTest.cpp

```diff
@@ -972,6 +972,417 @@ TEST_F(UprobesUnwindingVisitorSampleTest,
   EXPECT_EQ(discarded_samples_in_uretprobes_counter, 0);
 }
 
+TEST_F(UprobesUnwindingVisitorSampleTest, VisitStackSampleWillMakeUseOfUserSpaceStack) {
+  StackSamplePerfEvent event = BuildFakeStackSamplePerfEvent();
+
+  EXPECT_CALL(return_address_manager_, PatchSample).Times(1).WillOnce(Return());
+  EXPECT_CALL(maps_, Get).Times(1).WillOnce(Return(nullptr));
+
+  std::vector<unwindstack::FrameData> libunwindstack_callstack{kFrame1, kFrame2, kFrame3};
+
+  std::vector<StackSliceView> actual_stack_slices{};
+  EXPECT_CALL(unwinder_, Unwind(event.data.pid, nullptr, _, _, _, _))
+      .Times(1)
+      .WillOnce(DoAll(SaveArg<3>(&actual_stack_slices),
+                      Return(LibunwindstackResult{
+                          libunwindstack_callstack, {}, unwindstack::ErrorCode::ERROR_NONE})));
+
+  orbit_grpc_protos::FullCallstackSample actual_callstack_sample;
+  EXPECT_CALL(listener_, OnCallstackSample).Times(1).WillOnce(SaveArg<0>(&actual_callstack_sample));
+
+  std::vector<orbit_grpc_protos::FullAddressInfo> actual_address_infos;
+  auto save_address_info =
+      [&actual_address_infos](orbit_grpc_protos::FullAddressInfo actual_address_info) {
+        actual_address_infos.push_back(std::move(actual_address_info));
+      };
+  EXPECT_CALL(listener_, OnAddressInfo).Times(3).WillRepeatedly(Invoke(save_address_info));
+
+  std::atomic<uint64_t> unwinding_errors = 0;
+  std::atomic<uint64_t> discarded_samples_in_uretprobes_counter = 0;
+  visitor_.SetUnwindErrorsAndDiscardedSamplesCounters(&unwinding_errors,
+                                                      &discarded_samples_in_uretprobes_counter);
+
+  constexpr uint64_t kUserStackSize = 1024;
+  constexpr uint64_t kUserStackPointer = 16;
+  UprobesWithStackPerfEvent user_stack_event{
+      .timestamp = 10,
+      .data =
+          {
+              .stream_id = 1,
+              .pid = event.data.pid,
+              .tid = event.data.tid,
+              .regs =
+                  {
+                      .abi = 10,
+                      .sp = kUserStackPointer,
+                  },
+              .dyn_size = kUserStackSize,
+              .data = std::make_unique<char[]>(kUserStackSize),
+          },
+  };
+  PerfEvent{std::move(user_stack_event)}.Accept(&visitor_);
+
+  uint64_t dyn_size = event.data.dyn_size;
+  uint64_t sp = event.data.regs->sp;
+  PerfEvent{std::move(event)}.Accept(&visitor_);
+
+  EXPECT_THAT(actual_stack_slices,
+              ElementsAre(AllOf(Property(&StackSliceView::start_address, Eq(sp)),
+                                Property(&StackSliceView::size, Eq(dyn_size)),
+                                Property(&StackSliceView::data, NotNull())),
+                          AllOf(Property(&StackSliceView::start_address, Eq(kUserStackPointer)),
+                                Property(&StackSliceView::size, Eq(kUserStackSize)),
+                                Property(&StackSliceView::data, NotNull()))));
```

Comment:
Why not test `data` too? It should be easy to get the values.

Reply:
Same for the other tests

Reply:
At least I can check the pointers